### DISCUSSION TITLE
Zustand: typescript subscribe

### DISCUSSION THREAD

**Discussion by callmejm** [pmndrs/zustand#1822] [General]
How to use `subcribe` method in typescript ? I getting error `Property 'subscribe' does not exist on type 'BearState'`

```ts
//store.ts
import { create } from 'zustand'

interface BearState {
  bears: number
  increase: (by: number) => void
}

export const useBearStore = create<BearState>()((set) => ({
  bears: 0,
  increase: (by) => set((state) => ({ bears: state.bears + by })),
}))

```

```tsx
//App.tsx

function App() {
    const bearStore = useBearStore()
    const unsub = bearStore.subscribe(console.log)

    return (
        <div>
        </div>
    )
}

export default App

```

**Comment by dai-shi** [comment]
Your code looks well: https://tsplay.dev/N95Y8N

<img width="605" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/f9743fb2-98e6-4eb6-8d9e-61195a0cdc54">

**Reply by callmejm** [reply]
sorry, I have updated my question to more clear

**Reply by dai-shi** [reply]
You should just use the store api (attached to the hook) instead of the returned state. Also, useEffect would be necessary.
```js
useEffect(() => {
  const unsub = useBearStore.subscribe(console.log);
  return unsub;
}, []);
```

**Reply by callmejm** [reply]
still don't have susbcribe in useBearStore, all I got is only `bears` and `increase`

**Reply by callmejm** [reply]
Am I doing wrong ? 
```ts
export const useBearStore = create<BearState>()((set) => ({
  bears: 0,
  increase: (by) => set((state) => ({ bears: state.bears + by })),
}))
```

**Reply by callmejm** [reply]
How to use the store api (attached to the hook) instead of the returned state ? I cant find any website

**Reply by dai-shi** [reply]
Please check this https://tsplay.dev/mM7lQw and try to reproduce your error there.

**Reply by callmejm** [reply]
the link you provided is in the same file, but my case is in different file, and export useBearStore

**Reply by dai-shi** [reply]
So, it can be only reproduced if it's in a different file? Can you try putting in the same file and see if it works?

**Reply by callmejm** [reply]
I think I get what your meaning, I should use `useBearStore` instead of `bearStore`, so sorry for that
